### PR TITLE
Fix a bug introduced by the deletion of copy constructor of tensor

### PR DESCRIPTION
**Pull request by jerryzh168**

### CODE REVIEW

**Review by Yangqing — APPROVED** [review]
Accepting to unblock - thanks!

**Review by orionr — APPROVED** [review]

**Comment by orionr** [comment]
Looks like we might still have the segfault. @ezyang FYI

**Comment by ezyang** [comment]
Yes because elementwiseNot was not edited by this patch.

**Comment by ezyang** [comment]
I'd recommend reviewing every changed line from the original diff and checking if the problem applies.

File: caffe2/operators/elementwise_op_test.h [hunk 1]
```diff
@@ -59,7 +59,8 @@ void elementwiseAnd() {
     EXPECT_TRUE(op->Run());
     auto* blob = ws.GetBlob("Z");
     EXPECT_NE(nullptr, blob);
-    const auto& Z = blob->Get<caffe2::Tensor<Context>>();
+    caffe2::CPUContext context;
```
[the remark this answers is not in the document]
Comment (jerryzh168) [reply caffe2/operators/elementwise_op_test.h]: I just checked and that constructor will be removed in the next diff and the behavior will be Clone if we do `caffe2::Tensor Y(X);`. So let's keep it this way for less refactoring.

**Comment by jerryzh168** [comment]
@orionr Seems that we can only import to phabricator once?

**Comment by orionr** [comment]
@jerryzh168 does seem to be having issues. I'll unlink the diff and see if that works for now.

**Pull request closed without merging**